hal/inc/ble_hal.h

```diff
@@ -495,6 +495,15 @@ int hal_ble_stack_init(void* reserved);
  */
 int hal_ble_stack_deinit(void* reserved);
 
+/**
+ * Check if BLE stack is initialized.
+ *
+ * @param[in]   reserved    Reserved for future use.
+ *
+ * @returns     true if it is initialized, otherwise, false.
+ */
+bool hal_ble_is_initialized(void* reserved);
```

Comment:
Just a nit: for things like this perhaps might be a better idea from the get-go introduce a method to return an enum `state`, instead of one purpose `initialized` / `not initialized`.